crates/bindings-macro/src/table.rs

```diff
@@ -543,17 +547,46 @@ pub(crate) fn table_impl(mut args: TableArgs, mut item: MutItem<syn::DeriveInput
     let primary_col_id = primary_key_column.iter().map(|col| col.index);
     let sequence_col_ids = sequenced_columns.iter().map(|col| col.index);
 
+    let scheduled_reducer_type_check = args.scheduled.as_ref().map(|reducer| {
+        quote! {
+            spacetimedb::rt::scheduled_reducer_typecheck::<#original_struct_ident>(#reducer);
+        }
+    });
     let schedule = args
         .scheduled
         .as_ref()
         .map(|reducer| {
-            // scheduled_at was inserted as the last field
-            let scheduled_at_id = (fields.len() - 1) as u16;
-            quote!(spacetimedb::table::ScheduleDesc {
+            // better error message when both are missing
+            if scheduled_at_column.is_none() && primary_key_column.is_none() {
+                return Err(syn::Error::new(
+                    original_struct_ident.span(),
+                    "scheduled table missing required columns; add these to your struct:\n\
+                             #[primary_key]\n\
+                             #[auto_inc]\n\
+                             scheduled_id: u64,\n\
```

Comment:
Is `scheduled_id` required to have both of these attributes? In particular, I could imagine `auto_inc` not being required, though usually correct. (To be clear, I am not advocating for changing this message, just asking.)

Reply:
auto_inc isn't required, no